bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/DecimalValue.java

```diff
@@ -587,6 +455,15 @@ public static DecimalValue valueOf(long value) {
      * @return decimal value
      */
     public static DecimalValue valueOf(double value) {
+        if (Double.isNaN(value)) {
+            throw ErrorUtils.createInvalidDecimalError(NAN);
+        }
+        if (value == Double.POSITIVE_INFINITY) {
```

Comment:
Shall we add test for cover missing lines as well

Reply:
The lines shown as uncovered are already covered in the newly added negative tests. These lines are hitting when we run the tests locally. Not sure why it shows the warning.